docker/Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM python:3.8-slim-buster
+
+WORKDIR pypfopt
+COPY pyproject.toml poetry.lock ./
```

Comment:
Note that only the poetry files are copied initially. These change infrequently (compared to other code). Thus, the dependency installation remains cached by docker, even if other files are changed (much faster builds after the first time).